### ISSUE TITLE
stack_overview.svg needs to be updated to match the latest spec

### ISSUE THREAD

**Issue by ddorwin** [w3c/encrypted-media#2]
* needkey => encrypted
* isTypeSupported() => requestMediaKeySystemAccess(), createMediaKeys(), createSession()
* createSession() => generateRequest()

We should also add the new objects next to MediaKeySession.

**Comment by steelejoe** [comment]
I am working on this now.

**Comment by steelejoe** [comment]
The section labeled "Optional" does not appear to be optional -- I would expect an "encrypted" event to always be sent for an encrypted media. Is this not correct?

**Comment by steelejoe** [comment]
This is the updated SVG. Unfortunately generating a pull request failed for me -- have to dig more into that. [stack_overview.svg.zip](https://github.com/w3c/encrypted-media/files/86824/stack_overview.svg.zip)

**Comment by ddorwin** [comment]
Thanks, @steelejoe.

You are correct that firing "encrypted" is not optional. What is optional is whether the application handles it or provides data such that it fires before checking for supported types. Previously, checking the supported types was also optional, though advisable, previously. (The original spec said, "Likely only one of the arrows in the dotted box will occur.")

Now, requestMediaKeySystemAccess() is certainly required. The only thing that's optional is whether to handle "encrypted" and whether it occurs before or after any of the EME method calls. The dashed optional arrow is probably sufficient.

Nits:
* For requestMedia...KeySystemAccess(), use a hyphen like in a newspaper.
* The MediaKeySession boxes changed colors? Is this intentional? They should be distinct from other boxes.
* The far left arrow to Net should be a little shorter (it was wrong in the current version).
* The Generate Request arrows should be to the right of the generateRequest() arrows.
* There is something strange about the head of the second update() arrow. It looks like there might be two arrows there, one of which is offset.
* The first Generate Request arrow no longer touches the CDM box.

MediaKeySystemAccess and MediaKeys are not shown (and the arrows point to the MediaStack box) while MediaKeySession is. Adding those objects could complicate the image, so we may just want to leave it as is. One option might be to add them above the MediaKeySession boxes. MediaKeys would need to stretch to the other createSession() anyway.

**Comment by paulbrucecotton** [comment]
Can one of the EME editors (@ddorwin, @mwatson2 or @jdsmith3000) please review and commit Joe's changes?

**Comment by mwatson2** [comment]
The diagram looks good to me. The only thing that comes to mind is that it is also possible for a MediaKeySession to spontaneously generate a message and receive a response. This could happen for key renewal or for reporting the record of key usage. These flows are perhaps more important than the case of multiple sessions which is presently shown.

**Comment by steelejoe** [comment]
I thought about that. There are actually several use cases that are not represented by this diagram. Rather than clutter up the spec with all of them, what if we add supplemental diagrams to the Use Case wiki? It needs some love anyway to reflect the better breakdown of the "key rotation" use case(s).

**Comment by mavgit** [comment]
I suggest changing "Browser" to "User agent", which corresponds to the language in the spec. Both terms are used in the spec, but the term "User agent" is used consistently in the API descriptions a reader would be studying and wanting to follow the diagram.

**Comment by ddorwin** [comment]
@steelejoe's latest version looks good. I'll wait for him to reply in the PR before committing.

Regarding other use cases, this is just one non-normative example for illustration of the most basic license exchange (and showing that there can be multiple sessions). As with the code examples, I don't think we should try to cover all the use cases. That might be something covered in the primer we discussed a while ago or some similar documentation. Also, it's not clear that such uses would need the full stack - it may be best to just show the calls and event between the application and user agent.

@steelejoe, I do have the source document, but it does not have your edits, so it's probably best to work from the updated SVG or from scratch, especially if they are to focus on a subset of this image, as I suggested above.
